ACPs/113-provable-randomness/README.md

```diff
@@ -0,0 +1,165 @@
+
+	b. If the parent block does not have an empty `vrfSig` signature, the proposer would sign the bootStrappingBlockSignature with its BLS key. See the bootStrappingBlockSignature details below.
+
+2. The current proposer does not have a BLS key
+   
+   a. If the parent block has a non empty `vrfSig` signature, the proposer would set the proposed block `vrfSig` to the 32 byte hash result of the following preimage:
+	```
+	+-------------------------+----------+------------+
+	|  prefix :               | [8]byte  | "rng-derv" |
+	+-------------------------+----------+------------+
+	|  vrfSig :               | [96]byte |  96 bytes  |
+	+-------------------------+----------+------------+
+	```
+
+	b. If the parent block has an empty `vrfSig` signature, the proposer would leave the field empty.
+
+The bootStrappingBlockSignature that would be used above is the hash of the following preimage:
+
+```
++-----------------------+----------+------------+
+|  prefix :             | [8]byte  | "rng-root" |
++-----------------------+----------+------------+
+|  chainID :            | [32]byte |  32 bytes  |
++-----------------------+----------+------------+
+|  networkID:           | uint32   |  4 bytes   |
++-----------------------+----------+------------+
+```
+
+#### Step 3. Signature Verification
+
+This signature verification would perform the exact opposite of what was done in step 2, and would verify the cryptographic correctness of the operation.
+
+Validating the `vrfSig` would following this logic:
+1. The proposer has a BLS key
+
+	a. If the parent block had a non empty `vrfSig`, then a BLS signature verification of the proposed block  `vrfSig` would be made against the proposer’s BLS public key and the parent's block  `vrfSig` as the message.
+	b. If the parent block does has an empty `vrfSig`, then a BLS signature verification of the proposed block `vrfSig` against the proposer’s BLS public key and bootStrappingBlockSignature would take place.
+
+2. The proposer does not have a BLS key
+
+	a. If the parent block had a non empty `vrfSig`, then the hash of the preimage ( as described above ) would be compared against the proposed `vrfSig`.
+
+	b. If the parent block has an empty `vrfSig` then the proposer's `vrfSig` would be validated to be empty.
+
+#### Step 4. Extract the VRF Out and pass to block builders
+
+Calculating the VRF Out would be done by hashing the preimage of the following struct:
+
+```
++-----------------------+----------+------------+
+|  prefix :             | [8]byte  | "vrfout  " |
++-----------------------+----------+------------+
+|  vrfout:              | [96]byte |  96 bytes  |
++-----------------------+----------+------------+
+```
+
+Before calculating the VRF Out, the method needs to explicitly check the case where the `vrfSig` is empty. In that case, the output of the VRF Out needs to be empty as well.
+
+## Backwards Compatibility
```

Comment:
You mentioned it above already, but I think the `Backwards Compatibility` section should also call out that these changes necessitate a required network upgrade.

Reply:
np.